test/Performance/DynamoPerformanceTests/PerformanceTestFramework.cs

```diff
@@ -0,0 +1,158 @@
+            libraries.Add("Analysis.dll");
+            libraries.Add("GeometryColor.dll");
+        }
+
+        [ParamsSource(nameof(PerformanceTestSource))]
+        public string DynamoFilePath { get; set; }
+
+        #region Iteration setup and cleanup methods for Benchmarks
+
+        [IterationSetup(Target = nameof(OpenModelBenchmark))]
+        public void IterationSetupOpenModel()
+        {
+            Setup();
+        }
+
+        [IterationSetup(Target = nameof(RunModelBenchmark))]
+        public void IterationSetupRunModel()
+        {
+            Setup();
+
+            //open the dyn file
+            OpenModel(DynamoFilePath);
+        }
+
+
+        [IterationCleanup]
+        public void IterationCleanup()
+        {
+            Cleanup();
+        }
+
+        #endregion
+
+        #region Benchmark methods
+        /// <summary>
+        /// Automated creation of performance test cases.
+        /// </summary>
+        /// <param name="dynamoFilePath">The path of the dynamo workspace.</param>
+        [Benchmark]
+        public void OpenModelBenchmark()
+        {
+            //open the dyn file
+            OpenModel(DynamoFilePath);
+        }
+
+        [Benchmark]
+        public void RunModelBenchmark()
+        {
+            BeginRun();
+        }
+
+#endregion
+
+        /// <summary>
+        /// Populates the test cases based on DYN files in the performance tests folder.
+        /// </summary>
+        /// <returns></returns>
+        public static IEnumerable<string> PerformanceTestSource()
+        {
+            var fi = new FileInfo(Assembly.GetExecutingAssembly().Location);
+            string dir = fi.DirectoryName;
+
+            // Test location for all DYN files to be measured for performance 
+            // TODO: to be parameterized
+            string testsLoc = Path.Combine(dir, testDirectory);
+            var regTestPath = Path.GetFullPath(testsLoc);
+
+            var di = new DirectoryInfo(regTestPath);
+            var dyns = di.GetFiles("*.dyn");
+            foreach (var fileInfo in dyns)
+            {
+                yield return fileInfo.FullName;
+            }
+        }
+    }
+
+
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            // Running with an input dir location:
+            // DynamoPerformanceTests.exe "C:\directory path\"
+            // 
+            if (args.Length <= 0)
+            {
+                Console.WriteLine("Supply a path to a test directory containing DYN files");
+            }
+
+            // Use this call in order to run benchmarks on debug build of DynamoCore
+            //var summary = BenchmarkRunner.Run<PerformanceTestFramework>(
+            //new PerformanceTestFramework.AllowNonOptimized(testDir));
+
+            var summary = BenchmarkRunner.Run<PerformanceTestFramework>(
+                new PerformanceTestFramework.BenchmarkConfig(args[0]));
```

Comment:
I think we should eventually replace this with a more robust framework as we add more command line flags to this tool, but this is fine for now.

Reply:
I'm unclear right now what other options we can add. Perhaps add memory leak checks, etc. but didn't want to expand the scope of this work right now.